source/simulator.hpp

```diff
@@ -10,42 +15,174 @@
 #include "device/sender.hpp"
 #include "device/switch.hpp"
 #include "link.hpp"
+#include "logger/logger.hpp"
+#include "scheduler.hpp"
 
 namespace sim {
 
+using routing_table_t =
+    std::unordered_map<std::shared_ptr<IRoutingDevice>, std::shared_ptr<ILink>>;
```

Comment:
Let's use camel case as in all other projcet (sorry, `routing_table_t` was my suggestion, my miss)
```suggestion
using RoutingTable =
    std::unordered_map<std::shared_ptr<IRoutingDevice>, std::shared_ptr<ILink>>;
```